### PR TITLE
Warn if 'git://' protocol is used as the source of a pod

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/7705

/cc @KrauseFx @larsxschneider

### CODE REVIEW

File: lib/cocoapods/installer/pod_source_installer.rb [hunk 1]
```diff
@@ -110,18 +112,24 @@ def download_source
...
+      #
+      # @return [void]
       #
       def verify_source_is_secure(root_spec)
-        return if root_spec.source.nil? || root_spec.source[:http].nil?
-        http_source = URI(root_spec.source[:http])
-        return if http_source.scheme == 'https' || http_source.scheme == 'file'
-        UI.warn "'#{root_spec.name}' uses the unencrypted http protocol to transfer the Pod. " \
+        return if root_spec.source.nil? || (root_spec.source[:http].nil? && root_spec.source[:git].nil?)
+        source = if !root_spec.source[:http].nil?
```
Comment (dnkoutso) [inline lib/cocoapods/installer/pod_source_installer.rb]: anyone with more idiomatic ruby? suggestions welcomed

File: lib/cocoapods/installer/pod_source_installer.rb [hunk 2]
```diff
@@ -110,18 +112,24 @@ def download_source
...
+                   URI(root_spec.source[:http])
+                 elsif !root_spec.source[:git].nil?
+                   URI(root_spec.source[:git].to_s)
```
Comment (amorde) [inline lib/cocoapods/installer/pod_source_installer.rb]: Why does `:git` need `to_s` but `:http` does not?

Comment (dnkoutso) [reply lib/cocoapods/installer/pod_source_installer.rb]: @amorde good Q, for some of the tests and podspecs the `git` value resolves to a `Pathname`... I could also make the `:http` one to use `to_s` it should be harmless.

File: lib/cocoapods/installer/pod_source_installer.rb [hunk 3]
```diff
@@ -110,18 +112,24 @@ def download_source
...
+        return if root_spec.source.nil? || (root_spec.source[:http].nil? && root_spec.source[:git].nil?)
+        source = if !root_spec.source[:http].nil?
+                   URI(root_spec.source[:http].to_s)
+                 elsif !root_spec.source[:git].nil?
+                   URI(root_spec.source[:git].to_s)
+                 end
+        if UNENCRYPTED_PROTOCOLS.include?(source.scheme)
+          UI.warn "'#{root_spec.name}' uses the unencrypted '#{source.scheme}' protocol to transfer the Pod. " \
                 'Please be sure you\'re in a safe network with only trusted hosts in there. ' \
                 'Please reach out to the library author to notify them of this security issue.'
```
[the remark this answers is not in the document]
Comment (dnkoutso) [reply lib/cocoapods/installer/pod_source_installer.rb]: done

**Pull request merged**